### PR TITLE
Setup 23.12.0 branch

### PR DESCRIPTION
**Pull request by zoxpx**
**What this PR does / why we need it**:

This PR sets up the boilerplate code for the `px-rel-23.12.0` release, based off of `23.10.0`

**Which issue(s) this PR fixes** (optional)
Closes #  n/a

**Special notes for your reviewer**:

### CODE REVIEW

File: deploy/olm-catalog/portworx/23.12.0/portworx-certified.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,544 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: portworx-operator.v23.12.0
+  namespace: placeholder
+  annotations:
+    capabilities: Auto Pilot
+    categories: "Storage"
+    description: Cloud native storage solution for production workloads
+    containerImage: registry.connect.redhat.com/portworx/openstorage-operator@sha256:6c0d870d8573747f79cbebf41614258326131173668aa7be5638c8dd7846ea54
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.12.0/portworx-certified.clusterserviceversion.yaml:10]: *`portworx/px-operator:23.12.0-dev`

Same for the two instances of the image below.

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM

File: deploy/olm-catalog/portworx/23.12.0/portworx-certified.clusterserviceversion.yaml (hunk 1 above)
Comment (zoxpx) [reply deploy/olm-catalog/portworx/23.12.0/portworx-certified.clusterserviceversion.yaml:10]: Oh.. thanks!

**Comment by zoxpx** [comment]
Thanks for the review Piyush - merging the PR

**Pull request merged**